### ISSUE TITLE
`map_block()` with `new_axis` output has incorrect shape

### ISSUE THREAD

**Issue by dstansby** [dask/dask#11329]
**Describe the issue**:
When running `map_blocks()` with `new_axis` specified, the output shape of the dask array is not set correctly. In the below example I would expect it to be the same as the shape after computation.

**Minimal Complete Verifiable Example**:

```python
import dask.array as da
import numpy as np

def func(x):
    return np.stack([x, x + 0.5])

x = da.arange(6, chunks=2)
x_mapped = x.map_blocks(func, new_axis=[0])

print(x_mapped.shape)
# (1, 6)
print(x_mapped.compute().shape)
# (2, 6)
```

**Anything else we need to know?**:

**Environment**:

- Dask version: b7d9bf49f682de8d2ef51f4617e3da782400c290
- Python version: 3.12.3
- Operating System: macOS
- Install method (conda, pip, source): source

**Comment by phofl** [comment]
I don't think that we can guess the output shape with a high degree of fidelity. We should probably either set all chunks to NaN or force the specification of chunks.

**Comment by phofl** [comment]
Yeah, the interesting thing for Dask is the chunk sizes, the shape is just a convenient result from that, so we would need this information.

**Comment by phofl** [comment]
You can set chunks=(2, 2,) to fix that problem

**Comment by dstansby** [comment]
❤️ thanks for improving the docs for this!
